TM review

tmorley1 opened the pull request:
I have reviewed the document and made some changes.

- Corrected some typos/grammar
- Added in some pictures/screenshots to demonstrate certain points
- Added in QA branding (including QA logo, hero image, favicon)

Please can you check my changes before merge? Thank you!

tmorley1 committed:
* updating overview of chapter

* providing insight into purpose of initial version

* adding names of creators of documentation

* correcting some grammar/typos

tmorley1 committed:
Correcting typos/grammar

tmorley1 committed:
adding images to chapter

tmorley1 committed:
Adding alt text to image

tmorley1 committed:
* adding images

* correcting typos

tmorley1 committed:
adding design features:

* hero image
* QA logo
* favicon

rtadd-git commented on 04-overall-workflow.Rmd:
```diff
@@ -89,7 +91,7 @@ Repositories usually have either a "main" or a "master" branch. The two terms re
 
 - When you've made enough changes to complete your work item checkout the main 
-branch in your local repository do a pull in case anyone else had made changes.
+branch in your local repository and pull in case anyone else had made changes.
```
Would it be any clearer for a new user to explicitly say "git pull" rather than just "pull" here?

rtadd-git commented on index.Rmd:
```diff
@@ -1,32 +1,41 @@
 ## Overview
 
-This book aims act as a resource for analysts on how and why they should use version control.
+This book acts as a resource for analysts on how Git works and why it is useful for collaborative working and version control.
```
I think the double "and" in this sentence makes it a bit cumbersome. For me a comma before the first "and" would make it clearer.

rtadd-git commented on 04-overall-workflow.Rmd:
```diff
@@ -142,6 +147,8 @@ git push
 
 Tags effectively label specific Git commits so you can easily find them again in future. In DevOps they show up in the Repos section under Tags. They are valuable if you've produced some outputs and wanted to label the code which produced them for future reference. If you include "/" in your tag names, they will show up in folders on DevOps.
 
+![An example of using tags within a repository.](www/images/chapter4_tags.png)
```
To me this image looks like a product backlog image not an image of tags - let me know if you'd like to discuss the difference.

rtadd-git commented on 04-overall-workflow.Rmd:
```diff
@@ -55,6 +55,8 @@ git push -u origin --all <repo_URL>
 
 - Work may be assigned to you as a "Work Item". In DevOps use the link to make new branch. To correct a bug, prefix the branch name with "bug/". To add new functionality, prefix the branch name with "feature/".
 
+![Some example work items in DevOps.](www/images/chapter4_work_items.png)
```
This image doesn't seem quite right here to me. I think focussing too much on work items when this is about branches is likely to be confusing for some people.

It's perhaps slightly beyond the scope of your changes but could this section be clearer if we remove the work item content, or at least say how to make a branch without a work item too?

rtadd-git reviewed (changes requested):
This looks like a good set of changes thank you.
* I have left some comments inline, mostly small points of detail.
* I would also like to check where images are sourced from - do we have attribution for any taken from the internet, including existing images you didn't add?
* I think some of the old images are still in the www folder. Shall we remove the ones we're not using?

tmorley1 commented on 02-localproject.Rmd:
````diff
@@ -27,11 +27,11 @@ When you first install Git it is advised to perform some set up steps - they are
 First of all type
 ```sh
-git config --global user.name <Your Name>
+git config --global user.name "<Your Name>"
````
I was trying to follow this and I had to use both to get it to work, i.e. user.name "<Tom Morley>"

rtadd-git replied:
Interesting, that's not my experience. There's also an example here which doesn't have the <> in it: https://www.gitkraken.com/learn/git/git-config

Possibly an interesting one to ask others about?

tmorley1 committed:
- Making spelling/grammar corrections
- Removing names and adding QA Team email instead
- Adding sources to pictures
- Updating picture for tags
- Removing discussion on work items and making focus on creating new branches in DevOps

rtadd-git commented on 01-intro.Rmd:
```diff
@@ -42,7 +42,7 @@ At its most basic level Git allows the following:
 3. **Merge** - Systematically merge these changes back into the main version of the code.
 
-![](www/branching_diagram.png)
+![A diagram to demonstrate branching with Git.](www/images/chapter1_branching_diagram.png)
```
Is there a reference for this image like you have included with the others?

tmorley1 replied:
I didn't put this image in and a reverse image search doesn't give me any results for the source. Any ideas where this picture has come from?

rtadd-git commented on 04-overall-workflow.Rmd:
```diff
@@ -142,6 +147,8 @@ git push
 
 Tags effectively label specific Git commits so you can easily find them again in future. In DevOps they show up in the Repos section under Tags. They are valuable if you've produced some outputs and wanted to label the code which produced them for future reference. If you include "/" in your tag names, they will show up in folders on DevOps.
 
+![An example of using tags within a repository. Image taken from https://leonardo-matsumota.com](www/images/chapter4_tags.png)
```
I think this image has something in the foreground which isn't really about creating a tag and I'm concerned it might be off putting for a new user. I've emailed you an alternative.

rtadd-git reviewed (commented):
@tmorley1, thank you for fitting in the changes. I've made two more comments about images where they occur. In addition:
* There is a zip file that's appeared. What is this?
* Do you want to leave the username config as is?

tmorley1 committed:
- Altering tags image
- Deleting zip file

beatricegrace commented on 05-tips.Rmd:
```diff
@@ -45,7 +45,7 @@ Use
     git reset --hard <commit-hash>
 
-where <commit-hash> is the unique hash of the commit you want to go back to (you can find this via `git log`).
+where `<commit-hash>` is the unique hash of the commit you want to go back to (you can find this via `git log`).
```
I think same here as previous comment, this should probably be 'commit-hash' since the <> aren't needed

beatricegrace reviewed (changes requested):
Thanks Tom! I'm happy with your changes except that I agree with Rachel that we should remove the <> when we switch to using "" to avoid confusing people.

When I now knit this document, I get a weird switch in numbering where I get two chapter twos when I'm flicking through the file. The first is the 'Introduction' chapter which has a picture header and the second chapter is 'Version control using Git' which doesn't have a picture header. 

Is this the same for you too?

tmorley1 committed:
Removing < > signs for consistency in git code

tmorley1 committed:
removing < > signs

tmorley1 commented:
> 

I don't have two chapter twos, but I use the following commands rather than clicking 'Knit':
bookdown::render_book("index.Rmd")
browseURL("docs/index.html")

tmorley1 committed:
adding < > signs

beatricegrace reviewed (approved):
Thanks Tom, these changes all look great and I'm happy to approve. :)
